nodejs/lancedb/embedding/openai.ts

```diff
@@ -4,17 +4,41 @@
 import type OpenAI from "openai";
 import type { EmbeddingCreateParams } from "openai/resources/index";
 import { Float, Float32 } from "../arrow";
-import { EmbeddingFunction } from "./embedding_function";
+import { TextEmbeddingFunction } from "./embedding_function";
 import { register } from "./registry";
 
 export type OpenAIOptions = {
   apiKey: string;
   model: EmbeddingCreateParams["model"];
+  maxRetries?: number;
+  maxRequestsPerMinute?: number;
 };
 
+/**
+ * OpenAI embedding function with built-in rate limiting and retry support.
+ * 
+ * @example
+ * ```ts
+ * // Create with default rate limiting (0.9 requests per second) and retries (7 attempts)
+ * const embed = new OpenAIEmbeddingFunction({ 
+ *   model: "text-embedding-3-small",
+ *   apiKey: "your-api-key" 
+ * });
+ * 
+ * // Or customize rate limiting and retries
+ * const embed = new OpenAIEmbeddingFunction({ 
+ *   model: "text-embedding-3-small",
+ *   apiKey: "your-api-key",
+ *   maxRequestsPerMinute: 60, // 60 requests per minute
+ *   maxRetries: 5 // 5 retries with exponential backoff
+ * });
+ * 
+ * // Then use with retry and rate limiting
+ * const embeddings = await embed.computeSourceEmbeddingsWithRetry(["text1", "text2"]);
+ * ```
```

Comment:
What's the reason for making this a new public method? Why not allow retries in the existing `computeQueryEmbeddings` and `computeSourceEmbeddings` methods?